src/main/java/org/spongepowered/common/event/filter/delegate/GetterFilterSourceDelegate.java

```diff
@@ -0,0 +1,110 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.event.filter.delegate;
+
+import static org.objectweb.asm.Opcodes.ACONST_NULL;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.ARETURN;
+import static org.objectweb.asm.Opcodes.ASTORE;
+import static org.objectweb.asm.Opcodes.IFEQ;
+import static org.objectweb.asm.Opcodes.IFNE;
+import static org.objectweb.asm.Opcodes.INSTANCEOF;
+import static org.objectweb.asm.Opcodes.INVOKEINTERFACE;
+import static org.objectweb.asm.Opcodes.INVOKEVIRTUAL;
+
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
+import org.spongepowered.api.event.filter.Getter;
+import org.spongepowered.api.util.Tuple;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.Optional;
+
+public class GetterFilterSourceDelegate implements ParameterFilterSourceDelegate {
+
+    private final Getter anno;
+
+    public GetterFilterSourceDelegate(Getter a) {
+        this.anno = a;
+    }
+
+    @Override
+    public Tuple<Integer, Integer> write(ClassWriter cw, MethodVisitor mv, Method method, Parameter param, int local) {
+        Class<?> targetType = param.getType();
+        Class<?> eventClass = method.getParameterTypes()[0];
+        String targetMethod = this.anno.value();
+        Method targetMethodObj = null;
+        for (Method mth : eventClass.getMethods()) {
+            if (targetMethod.equals(mth.getName())) {
+                if (targetMethodObj != null) {
+                    throw new IllegalArgumentException("Multiple matches found for getter " + targetMethod + " in type " + eventClass.getName());
+                }
+                if (mth.getParameterCount() != 0) {
+                    throw new IllegalArgumentException(
+                            "Method " + mth.toGenericString() + " specified by getter annotation has non-zero parameter count");
+                }
+                if (!mth.getReturnType().equals(Optional.class) && !mth.getReturnType().equals(targetType)) {
```

Comment:
What if instead of the specific `equals` for the target type, perhaps where the target type is checked to be `isAssignableFrom`? This would allow subclass checks as well.